src/jit/decomposelongs.cpp

```diff
@@ -878,27 +878,17 @@ GenTree* DecomposeLongs::DecomposeNeg(LIR::Use& use)
     GenTree* gtLong = tree->gtGetOp1();
     noway_assert(gtLong->OperGet() == GT_LONG);
 
-    LIR::Use op1(Range(), &gtLong->gtOp.gtOp1, gtLong);
-    op1.ReplaceWithLclVar(m_compiler, m_blockWeight);
-
-    LIR::Use op2(Range(), &gtLong->gtOp.gtOp2, gtLong);
-    op2.ReplaceWithLclVar(m_compiler, m_blockWeight);
-
-    // Neither GT_NEG nor the introduced temporaries have side effects.
-    tree->gtFlags &= ~GTF_ALL_EFFECT;
     GenTree* loOp1 = gtLong->gtGetOp1();
     GenTree* hiOp1 = gtLong->gtGetOp2();
 
     Range().Remove(gtLong);
+    Range().Remove(tree);
 
-    GenTree* loResult    = tree;
-    loResult->gtType     = TYP_INT;
-    loResult->gtOp.gtOp1 = loOp1;
+    GenTree* loResult = m_compiler->gtNewOperNode(GT_NEG, TYP_INT, loOp1);
```

Comment:
When possible, we prefer to reuse existing nodes in order to decrease memory pressure. Please leave the original code that reused the node.